src/js/angular/core/directives/yasgui-component/yasgui-component.directive.js

```diff
@@ -343,24 +354,63 @@ function yasguiComponentDirective(
                             onQueryAborted: onQueryAborted,
                             beforeUpdateQuery: getBeforeUpdateQueryHandler()
                         };
-                        angular.extend(config, $scope.config || DEFAULT_CONFIG, $scope.yasguiConfig);
+
+                        angular.extend(config, DEFAULT_CONFIG, $scope.yasguiConfig);
+
                         $scope.ontotextYasguiConfig = config;
 
                         if (angular.isFunction($scope.afterInit)) {
                             $scope.afterInit();
                         }
+
+                        addDirtyCheckHandlers();
                     }
                 }));
 
+            const addDirtyCheckHandlers = () => {
+                const waitOntotextInitialized = $interval(function () {
+                    const ontotextYasguiElements = $scope.getOntotextYasguiElements();
+                    if (ontotextYasguiElements) {
+                        ontotextYasguiElements.on('paste.sparqlQuery', '.CodeMirror', function () {
```

Comment:
You can extract this callback function to reduce the nesting and improve readability.

Reply:
Done